From 712281412a19f9845dac2eeabd341fcce7a8cfbf Mon Sep 17 00:00:00 2001
From: Matthew Schwartz <matthew.schwartz@linux.dev>
Date: Mon, 3 Aug 2026 15:16:11 -0700
Subject: [PATCH 183/188] platform/x86: msi-wmi-platform: terminate msi_quirks
 DMI table

The msi_quirks table lacks a terminating empty entry so it crashes the
kernel on non-quirked MSI Claw devices.

Signed-off-by: Matthew Schwartz <matthew.schwartz@linux.dev>
---
 drivers/platform/x86/msi-wmi-platform.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/platform/x86/msi-wmi-platform.c b/drivers/platform/x86/msi-wmi-platform.c
index 2439b4f00..fdc2f6f4c 100644
--- a/drivers/platform/x86/msi-wmi-platform.c
+++ b/drivers/platform/x86/msi-wmi-platform.c
@@ -268,6 +268,7 @@ static const struct dmi_system_id msi_quirks[] = {
 		},
 		.driver_data = &quirk_gen2,
 	},
+	{ }
 };
 
 static int msi_wmi_platform_parse_buffer(union acpi_object *obj, u8 *output, size_t length)
-- 
2.55.0

